Controllers/TicketsController.cs

```diff
@@ -0,0 +1,142 @@
+using GeekHub.Attributes;
+using GeekHub.Data;
+using GeekHub.DTOs.Ticket;
+using GeekHub.Models;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.JsonPatch;
+using Microsoft.AspNetCore.JsonPatch.Exceptions;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Serilog;
+
+namespace GeekHub.Controllers
+{
+    [Authorize]
+    [AddUserId]
+    [ApiController]
+    [Route("Projects/{projectId:guid}/[controller]")]
+    public class TicketsController : ControllerBase
+    {
+        private readonly ApplicationDbContext _dbContext;
+        private readonly IMapper _mapper;
+        private readonly IAuthorizationService _authorizationService;
+
+        public TicketsController(ApplicationDbContext dbContext, IMapper mapper,
+            IAuthorizationService authorizationService)
+        {
+            _dbContext = dbContext;
+            _mapper = mapper;
+            _authorizationService = authorizationService;
+        }
+
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<TicketResponseDto>>> ReadProjectTickets(
+            [FromRoute] Guid projectId
+        )
+        {
+            try
+            {
+                var tickets = await _dbContext.Tickets.Include(t => t.Project).Where(t => t.Project.Id == projectId)
+                    .ToListAsync();
+
+                return Ok(_mapper.Map<IEnumerable<TicketResponseDto>>(tickets));
+            }
+            catch (Exception exception)
+            {
+                Log.Fatal(exception, "Fatal error while reading project tickets");
+                return Problem();
+            }
+        }
+
+        [HttpPost]
+        public async Task<ActionResult<TicketResponseDto>> CreateProjectTicket(
+            [FromRoute] Guid projectId,
+            [FromBody] CreateTicketRequestDto createTicketRequestDto
+        )
+        {
+            try
+            {
+                var project = await _dbContext.Projects.Include(p => p.Users)
+                    .SingleOrDefaultAsync(p => p.Id == projectId);
+                if (project == default(Project)) return NotFound();
+
+                var authorization =
+                    await _authorizationService.AuthorizeAsync(User, project, Permission.CreateTicket.ToString());
+                if (!authorization.Succeeded) return Forbid();
+                
+                var ticket = _mapper.Map<Ticket>(createTicketRequestDto);
+                if (ticket.ParentTicketId is not null &&
+                    await _dbContext.Tickets.FindAsync(ticket.ParentTicketId) is null) return NotFound();
+                if (ticket.ReporterId is not null && project.Users.SingleOrDefault(u => u.Id == ticket.ReporterId) ==
+                    default(ApplicationUser)) return BadRequest();
+                if (ticket.AssigneeId is not null && project.Users.SingleOrDefault(u => u.Id == ticket.AssigneeId) ==
+                    default(ApplicationUser)) return BadRequest();
```

Comment:
I'm a bit confused as to how to read this; is it that `ReporterId` and `AssigneeId`, if they exist, must point to default `ApplicationUser`?

Reply:
If a Reporter/Assignee is provided they must exist inside the project, if the default of `ApplicationUser` is returned, it means they do not exist

Reply:
Ohhhhh okay right.